### PR TITLE
Add partial support for Proactive rules

### PR DESCRIPTION
**Pull request by bmorrissirromb**
*Issue #, if available:*
Part of #400 

*Description of changes:*
This partially implements issue #400. Config does not support Proactive evaluation modes for Organization rules. Only single-account rules support EvaluationMode arguments.

So, this change allows single-account (non-Organization rules) rules to use proactive evaluation modes.

This will be lumped into the 0.18.0 release. I am not planning to do a 0.17.12 release.

The CLI will have two new options:
- `rdk create` will support an argument for `--evaluation-mode` and store it in the `parameters.json` file
- `rdk deploy` (but not `deploy-organization`) will this feature by reading `EvaluationMode` from `parameters.json`.

In addition to this change, there are a few minor changes lumped in here:
- CFTs are now maintained as YAMLs, not JSONs, because YAML is just better.
- Carlo and Nima are listed as active maintainers. JDJ is moved to a former maintainer.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: rdk/template/configRuleOrganization.yaml [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+  BoundaryPolicyArn:
+    Description: ARN of a Boundary Policy, will be used only if LambdaRoleArn is NOT set.
+    Type: String
+    Default: ""
+  SourceBucket:
+    Description: Name of the S3 bucket that you have stored the rule zip files in.
+    Type: String
+    MinLength: "1"
+    MaxLength: "255"
+  # SourcePath:
```
Comment (depaolism) [inline rdk/template/configRuleOrganization.yaml:35]: Why was this commented out?

File: rdk/rdk.py [hunk 2]
```diff
@@ -3450,6 +3475,12 @@ def __parse_rule_args(self, is_required):
                 print("Rule names must be 128 characters or fewer.")
                 sys.exit(1)
 
+        if self.args.evaluation_mode in ["PROACTIVE", "BOTH"] and self.args.resource_types is None:
+            print(
+                "You are attempting to create a proactive rule without a configuration-change trigger. This is not supported. Please revise your request."
+            )
+            exit(1)
```
Comment (depaolism) [inline rdk/rdk.py:3482]: Should this be `sys.exit(1)`?

File: rdk/template/configRuleOrganization.yaml (hunk 1 above)
Comment (depaolism) [reply rdk/template/configRuleOrganization.yaml:35]: Also commented out in 2 additional locations in the code base.

File: rdk/__init__.py
[the remark this answers is not in the document]
Comment (depaolism) [reply rdk/__init__.py:1]: This was updated.

File: rdk/template/configRuleOrganization.yaml (hunk 1 above)
Comment (bmorrissirromb) [reply rdk/template/configRuleOrganization.yaml:35]: It was not used in the CFT anywhere, so I think it's just bloated old code that we can eventually get rid of.

File: rdk/rdk.py (hunk 2 above)
Comment (bmorrissirromb) [reply rdk/rdk.py:3482]:
```suggestion
            sys.exit(1)
```

**Review by depaolism — APPROVED** [review]

**Pull request merged**